spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/mapping/DefaultNeo4jConverter.java

```diff
@@ -470,10 +471,15 @@ public Object getParameterValue(PreferredConstructor.Parameter parameter) {
 			for (Value relatedEntity : list.asList(Function.identity())) {
 				Neo4jPersistentProperty idProperty = concreteTargetNodeDescription.getRequiredIdProperty();
 
+				String relatedEntityIdKey = Optional.ofNullable(concreteTargetNodeDescription.getIdDescription())
+					.flatMap(IdDescription::getOptionalGraphPropertyName)
+					.orElse(idProperty.getName());
+
 				// internal (generated) id or external set
 				Object idValue = idProperty.isInternalIdProperty()
 					? relatedEntity.get(NAME_OF_INTERNAL_ID)
-					: relatedEntity.get(idProperty.getName());
+					: relatedEntity.get(relatedEntityIdKey);
+
```

Comment:
I cannot add this as a suggestion via GH, so please do this instead after adding the above suggestion:

```
				Object idValue = relatedEntity.get(relatedEntityIdKey);
```

Reason for pulling the condition up: See above.